### ISSUE TITLE
Google Analytics is not configured for SPA

### ISSUE THREAD

**Issue by TimidRobot** [creativecommons/cccatalog-frontend#116]
- 2018-09-23: cccatalog-frontend was updated to be a single page app (SPA) and was deployed without Google Analytics.
- 2018-10-01: Google Analytics was added back, but it was configured for a traditional site with full page loads.
- Now: cccatalog-frontend needs to be updated for Single Page Application Tracking per [Single Page Application Tracking  |  Analytics for Web (analytics.js)  |  Google Developers](https://developers.google.com/analytics/devguides/collection/analyticsjs/single-page-applications).

**Comment by kgodey** [comment]
@TimidRobot the changes @aldenstpage and I made do not affect search, if that's an issue, I'll have to fix that separately. What we did sends Google Analytics a new pageview when someone navigates to a single image's detail page, which previously wasn't getting counted a new pageview.

If you look at the "CC Search Beta (Prod)" project in Google Analytics and navigate to Behavior > Site Content ([this link may work](https://analytics.google.com/analytics/web/#/report/content-pages/a2010376w131694782p135658737/)), you can see the uptick in page views starting Dec 10. You should also be able to see a `/photos/{uuid}` URL in the list of top pages, GA was not registering any `/photos/` pages before.

If there's an additional search issue to fix, could you link me to what you're looking at in Google Analytics and what you expect to see?

**Comment by kgodey** [comment]
Thanks for the info!

**Comment by zackkrida** [comment]
Hi there,

I saw this issue and wanted to chime in—there is a `to.fullPath()` method that would be useful in this case. The whole `router.afterEach()` call would look like this:

```js
router.afterEach((to) => {
  if (to && to.query && to.query.q) {
    store.commit(SET_QUERY, { query: to.query, override: true });
  }
  store.commit(SET_IMAGE, { image: {} });
  store.commit(SET_IMAGES, { images: [] });

  /* global ga */
  ga('set', 'page', to.fullPath);
  ga('send', 'pageview');
});
```

This sends google the full path, including any query params: 

![screen shot 2018-12-17 at 3 35 29 pm](https://user-images.githubusercontent.com/6351754/50113998-7c065480-0211-11e9-8877-f7a4eef21800.png)

This us useful for non-search pageviews as well—any utm parameters from marketing emails, etc. will be captured in each pageview. 😸

**Comment by kgodey** [comment]
This is fixed by #135.
